lib/resource-mapper.js

```diff
@@ -69,21 +74,41 @@ class ResourceMapper {
   }
 
   // Gets the base URL for the given hostname
-  getBaseUrl (hostname) {
-    return !this._includeHost ? this._rootUrl
+  // Optionally, a pathname may be passed that will be appended to the baseUrl.
+  getBaseUrl (hostname, pathname) {
```

Comment:
That method might be confusingly named then.
How about `resolveUrl (hostname, pathname = '')` or so?